src/mon/MonClient.cc

```diff
@@ -74,36 +73,36 @@ MonClient::~MonClient()
 {
   delete auth_supported;
   delete session_established_context;
-  delete auth;
+  auth.reset();
```

Comment:
No need.  unique_ptr will delete its contained pointer in its own destructor (which will be called along with the destructors for all other MonClient members after the body of ~MonClient()).  That's the point of RAII!